environment/src/main/java/com/sequenceiq/environment/environment/validation/securitygroup/gcp/GcpEnvironmentSecurityGroupValidator.java

```diff
@@ -70,18 +76,26 @@ private void validateSecurityGroup(EnvironmentDto environmentDto, ValidationResu
         CloudSecurityGroups securityGroups = platformParameterService.getSecurityGroups(request);
 
         boolean securityGroupFoundInRegion = false;
-        if (Objects.nonNull(securityGroups.getCloudSecurityGroupsResponses())
-                && Objects.nonNull(securityGroups.getCloudSecurityGroupsResponses().get(region.getName()))) {
-            for (CloudSecurityGroup cloudSecurityGroup : securityGroups.getCloudSecurityGroupsResponses().get(region.getName())) {
-                String groupId = cloudSecurityGroup.getGroupId();
-                if (groupId.equalsIgnoreCase(securityGroupId)) {
-                    securityGroupFoundInRegion = true;
-                    break;
+        Set<CloudSecurityGroup> cloudSecurityGroups = securityGroups.getCloudSecurityGroupsResponses().get(region.getName());
+        for (String securityGroupId : getSecurityGroupIds(securityGroupIds)) {
+            securityGroupFoundInRegion = false;
+            if (Objects.nonNull(securityGroups.getCloudSecurityGroupsResponses())
+                    && Objects.nonNull(cloudSecurityGroups)) {
```

Comment:
As mentioned above, the `Objects.nonNull(securityGroups.getCloudSecurityGroupsResponses())` condition is ineffective / too late here, it should be moved before the first usage of that reference.